src/events/balancer.py

```diff
@@ -159,6 +173,17 @@ def _on_config_changed(self, _: EventBase) -> None:
                 )
                 content_changed = True
 
+        # On k8s, adding/removing a broker does not change the bootstrap server property if exposed by nodeport
+        broker_capacities = self.charm.state.balancer.broker_capacities
+        if (
+            file_content := json.loads(
+                "".join(self.workload.read(self.workload.paths.capacity_jbod_json))
+            )
+        ) != broker_capacities:
+            logger.info(f"Balancer {self.charm.unit.name.split('/')[1]} updating capacity config")
+
+            content_changed = True
+
```

Comment:
**question:** What was the reason this wasn't part of the `changed_map` above? Because we wouldn't be able to log out the actual change?

If so, we could add an extra string item to each tuple for a unique log. Then:

```python
for kind, path, state_content, log in changed_map:
    ....
    if log:
        logger.info(log)
    else:
        logger.info(  # as normal
```

Reply:
Ah, yes, I went for simplicity here because the changed map deals with lines in a flat file (and with a .properties structure, meaning one `key=value` per line), but the capacity file requires more work to display the changes. I'll think of something

Reply:
I wrote a simple helper in 310be121632d to display the diff in the logs, let me know what you think